test/e2e/azure_test.go

```diff
@@ -603,4 +604,192 @@ var _ = Describe("Workload cluster creation", func() {
 
 		})
 	})
+	Context("CSI=internal CCM=internal AzureDiskCSIMigration=false: upgrade to v1.23 [OPTIONAL]", func() {
+		It("should create volumes dynamically with intree cloud provider", func() {
+			By("Creating workload cluster v1.22 using user-assigned identity")
+			clusterName = getClusterName(clusterNamePrefix, "intree-providers")
+			configCluster := defaultConfigCluster(clusterName, namespace.Name)
+			configCluster.KubernetesVersion = e2eConfig.GetVariable(PreCSIKubernetesVer)
+			configCluster.WorkerMachineCount = pointer.Int64Ptr(1)
+			configCluster.Flavor = "user-assigned-managed-identity"
```

Comment:
Can we ensure that the cluster we upgrade has 3 control plane nodes so we have test coverage for the HA scenario?

Reply:
Sure. Will do. 
Out of curiosity -- What is the HA scenario here? Like these three tests don't have any.

Reply:
3 control plane nodes

The most common way that we do that now is by setting this in the `ConfigCluster` of the test: `ControlPlaneMachineCount: pointer.Int64Ptr(3),`

Reply:
I see. I set the nodes to 1 intentionally as I do not know what is the benefit of having 3 control plane nodes. 
I am trying to ask myself : If I set the node count to 3, what is that extra thing( if HA, what is it) that gets validated by this upgrade e2e?

Reply:
Sure. Thanks. I will update the PR to have 3 CP nodes when we upgrade.